core/src/main/scala/flatgraph/DiffGraphApplier.scala

```diff
@@ -408,7 +408,7 @@ private[flatgraph] class DiffGraphApplier(graph: Graph, diff: DiffGraphBuilder)
     val oldQty = graph.neighbors(pos).asInstanceOf[Array[Int]]
     val edgeProp = graph.neighbors(pos + 2) match {
       case _: DefaultValue => graph.schema.allocateEdgeProperty(nodeKind, direction, edgeKind, size)
-      case other           => other
+      case other: Array[_] => other.clone() // this is not strictly necessary, but everything else is copy-on-write
```

Comment:
I think we should remove this copy and replace it with a comment that we will need to update the code once we switch away from full copy-on-write.

If we're eating the extra allocs for graph updates, can as well reap the benefits in terms of faster clone.

Reply:
yup that's what I meant with my comment in https://github.com/joernio/flatgraph/pull/163#issuecomment-2016856624

I removed the clone again and added a comment.